include/mbedtls/ssl.h

```diff
@@ -620,7 +620,7 @@
 #define MBEDTLS_TLS_EXT_SIG_ALG_CERT                50 /* RFC 8446 TLS 1.3 */
 #define MBEDTLS_TLS_EXT_KEY_SHARE                   51 /* RFC 8446 TLS 1.3 */
 
-#if MBEDTLS_SSL_DTLS_CONNECTION_ID_COMPAT == 0
+#if defined(MBEDTLS_SSL_DTLS_CONNECTION_ID_COMPAT) && MBEDTLS_SSL_DTLS_CONNECTION_ID_COMPAT == 0
```

Comment:
Before this change, if `MBEDTLS_SSL_DTLS_CONNECTION_ID_COMPAT` was not defined, `MBEDTLS_TLS_EXT_CID` would get 254.

With this change, if `MBEDTLS_SSL_DTLS_CONNECTION_ID_COMPAT` is undefined, `MBEDTLS_TLS_EXT_CID` will get 54.

That seems wrong for a change designed to just workaround a `-Wundef` warning

Reply:
not that it matter much, but you're right. fixed.